### DISCUSSION TITLE
Feature Request: Local Authentication

### DISCUSSION THREAD

**Discussion by bitmvk** [flet-dev/flet#5226] [Extensions]
**Please Describe The Feature To Be Added**

local authentication when delicate data is to be shown. Like in a Credit Card App.

***local authentication includes:***  Face Id, Fingerprint, Passcode

**Suggest A Solution**

https://pub.dev/packages/local_auth

**Comment by benson-nderitu** [comment]
Is this issue still on track? How is the progress?

**Reply by benson-nderitu** [reply]
Ideal Features:

- Simple API: _local_auth.authenticate()_
- Support for biometrics (fingerprint, Face ID)
- Fallback to device password/PIN if biometrics are unavailable
- Error handling (authentication failed, user cancelled)
- Cross-platform support
- Event-based or async API for good UX


<hr>
<h3><code inline="">local_auth.authenticate()</code> — Suggested Parameters</h3>

Parameter | Type | Description
-- | -- | --
reason/message | str | Text shown to the user explaining why authentication is needed.
biometric_only | bool | If True, disables fallback to PIN/password. Default: False.
sensitive | bool | If True, enables enhanced security handling.
allow_fallback | bool | Allow fallback to alternative auth (e.g., device PIN). Default: True.
timeout | int | Max time (in seconds) to wait for user interaction. Optional.
platforms | list[str] | Limit which platforms the auth is enabled on (e.g., ["android", "ios"]).
on_success | Callable | Callback if authentication succeeds. Optional.
on_error | Callable | Callback for handling failure or cancellation. Optional.
